src/schema/posts.ts

```diff
@@ -1582,6 +1635,91 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
     },
   },
   Mutation: {
+    createSourcePostModeration: async (
+      _,
+      {
+        sourceId,
+        image,
+        title,
+        content,
+        commentary,
+        type,
+        sharedPostId,
+        imageUrl,
+        externalLink,
+      }: CreateSourcePostModerationArgs,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      const { con, userId } = ctx;
+
+      const sourceMember = await con.getRepository(SourceMember).findOne({
+        where: { sourceId: sourceId, userId: userId },
+        relations: ['source'],
+      });
+
+      const source = await sourceMember?.source;
+
+      if (!source) {
+        throw new ForbiddenError('Access denied!');
+      }
```

Comment:
Doesn't `canAccessSource` handle this already?

Reply:
+1

also do we want to check type of source as well, and throw if its not a squad, just to make sure no accidental moderations for sources if someone is playing with API directly, you can see example in `joinSource`

Reply:
@rebelchris it doesn't because source is not an optional parameter